questionnaire/submissions-service.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const createMessageBusCaller = require('../services/message-bus');
+const createQuestionnaireDAL = require('./questionnaire-dal');
+
+function createSubmissionService({logger}) {
+    const db = createQuestionnaireDAL({logger});
+
+    async function getQuestionnaireIdsBySubmissionStatus(status) {
+        return db.getQuestionnaireIdsBySubmissionStatus(status);
+    }
+
+    async function updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus) {
+        return db.updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus);
+    }
+
+    async function postSubmissionsByStatus(status) {
+        const submissionStatus = status.toUpperCase();
+        const questionnaireIds = await getQuestionnaireIdsBySubmissionStatus(submissionStatus);
+        let response = [];
+        if (questionnaireIds.length) {
+            const messageBusCaller = createMessageBusCaller({logger});
+            const promises = questionnaireIds.map(async id => {
+                await updateQuestionnaireSubmissionStatus(id, 'IN_PROGRESS');
```

Comment:
should this be await? I am thinking it should. having this be set to IN_PROGRESS first will avoid it being picked up by a subsequent request.

Reply:
It looks like this could get stuck "IN_PROGRESS" should the message bus call fail

Reply:
I'm trying to follow the flow for this, if  the application service was down, I understand they would queue up and be processed eventually, but if the message bus was down would these promises not just return an error, leaving these cases `IN_PROGRESS` and unable to get picked up by a later job?

Reply:
are you basically saying that we probably need a try/catch safety net to avoid that situation? Something like the below.

````
const promises = questionnaireIds.map(async id => {
                await updateQuestionnaireSubmissionStatus(id, 'IN_PROGRESS');
                let promise;
                try {
                    promise = messageBusCaller.post('SubmissionQueue', {
                        applicationId: id
                    });
                } catch (err) {
                    await updateQuestionnaireSubmissionStatus(id, 'FAILED');
                }
                return promise;
            });
````